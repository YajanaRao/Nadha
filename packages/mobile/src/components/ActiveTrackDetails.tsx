import React from 'react';
import {View, StyleSheet, Dimensions} from 'react-native';
import FastImage from 'react-native-fast-image';
import {Subheading, Title} from 'react-native-paper';
import {DefaultImage} from './DefaultImage';
import {TrackProps} from '../types';

interface Props {
  track: TrackProps;
}

export const ActiveTrackDetails = ({track}: Props) => {
  return (
    <View style={{flex: 1}}>
      <View style={styles.centerContainer}>
        {track.cover ? (
          <FastImage source={{uri: track.cover}} style={[styles.artCover]} />
        ) : (
          <DefaultImage style={styles.artCover} />
        )}
      </View>
      <View style={styles.centerContainer}>
        <Title numberOfLines={1}>{track.title}</Title>
        <Subheading numberOfLines={1}>
          {track.artist ? track.artist : track.album}
        </Subheading>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  artCover: {
    borderRadius: 12,
    elevation: 4,
    height: Dimensions.get('window').width - 50,
    maxHeight: 300,
    maxWidth: 300,
    width: Dimensions.get('window').width - 50,
  },
  centerContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 16,
    marginVertical: 8,
  },
});
