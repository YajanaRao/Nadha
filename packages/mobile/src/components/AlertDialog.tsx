import React from 'react';
import {Button, Paragraph, Dialog, Portal} from 'react-native-paper';

interface AlertProps {
  visible: boolean;
  title: string;
  message: string;
  action(): void;
  hideDialog(): void;
}

export const AlertDialog = ({
  visible,
  title,
  message,
  action,
  hideDialog,
}: AlertProps) => {
  return (
    <Portal>
      <Dialog visible={visible} onDismiss={hideDialog}>
        <Dialog.Title>{title}</Dialog.Title>
        <Dialog.Content>
          <Paragraph>{message}</Paragraph>
        </Dialog.Content>
        <Dialog.Actions>
          <Button onPress={action}>Ok</Button>
          <Button onPress={hideDialog}>Cancel</Button>
        </Dialog.Actions>
      </Dialog>
    </Portal>
  );
};
