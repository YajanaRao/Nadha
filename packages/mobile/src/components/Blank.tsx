import React from 'react';
import {Button, useTheme} from 'react-native-paper';
import {StyleSheet, View} from 'react-native';
import {Tree} from './Tree';

interface Props {
  text: string;
  fetchData(): void;
}

export const Blank = ({text, fetchData}: Props) => {
  const theme = useTheme();
  const {colors} = theme;
  return (
    <View style={[styles.container, {backgroundColor: colors.background}]}>
      <Tree message={text} />
      <View style={styles.padding}>
        <Button icon="refresh" mode="text" onPress={fetchData}>
          Refresh
        </Button>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  padding: {
    margin: 4,
  },
});
