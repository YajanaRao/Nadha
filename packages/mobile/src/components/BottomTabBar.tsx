import React from 'react';
import {View, StyleSheet} from 'react-native';
import {Caption, TouchableRipple, Surface, Divider} from 'react-native-paper';
import {PlayerBarContainer} from '../containers/PlayerBarContainer';
import {NavigationState} from '@react-navigation/core';

interface BottomTabBarProps {
  renderIcon(route: NavigationState, focused: boolean): void;
  getLabelText(route: NavigationState): void;
  onTabPress(route: NavigationState): void;
  onTabLongPress(route: NavigationState): void;
  getAccessibilityLabel(route: NavigationState): void;
  state: NavigationState;
}

export const BottomTabBar = ({
  renderIcon,
  getLabelText,
  onTabPress,
  onTabLongPress,
  // getAccessibilityLabel,
  state,
}: BottomTabBarProps) => {
  const {routes, index} = state;
  return (
    <Surface style={{elevation: 4}}>
      <PlayerBarContainer />
      <Divider />
      <Surface style={[styles.container]}>
        {routes.map((route: NavigationState, routeIndex: number) => {
          const isRouteActive = routeIndex === index;
          // const color = isRouteActive ? "green" : "red";
          return (
            <TouchableRipple
              key={route.key}
              style={styles.tabButton}
              // rippleColor={colors.primary}
              // underlayColor={colors.primary}
              borderless
              onPress={() => {
                onTabPress({route});
              }}
              onLongPress={() => {
                onTabLongPress({route});
              }}
              // accessibilityLabel={getAccessibilityLabel({ route })}
            >
              <View style={{justifyContent: 'center', alignItems: 'center'}}>
                {renderIcon({route, focused: isRouteActive})}
                <Caption
                  style={{
                    textAlign: 'center',
                    fontSize: 10,
                    margin: 0,
                    padding: 0,
                  }}>
                  {getLabelText({route})}
                </Caption>
              </View>
            </TouchableRipple>
          );
        })}
      </Surface>
    </Surface>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 4,
  },
  tabButton: {
    flex: 1,
    justifyContent: 'center',
    margin: 0,
    padding: 0,
    alignItems: 'center',
  },
});
