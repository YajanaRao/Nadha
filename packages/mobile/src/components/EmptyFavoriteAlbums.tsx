import React from 'react';
import {Title, Subheading} from 'react-native-paper';
import {View, ViewStyle} from 'react-native';
import {ThemedIcon} from './ThemedIcon';

const CONTAINER: ViewStyle = {
  alignItems: 'center',
  flex: 1,
  justifyContent: 'center',
  margin: 24,
};

export const EmptyFavoriteAlbums = () => {
  return (
    <View style={CONTAINER}>
      <ThemedIcon name="heart-outline" size={80} />
      <Title>No favorites yet.</Title>
      <Subheading style={{textAlign: 'center'}}>
        Songs, albums, artists and playlists you&apos;ve liked will live here
      </Subheading>
    </View>
  );
};
