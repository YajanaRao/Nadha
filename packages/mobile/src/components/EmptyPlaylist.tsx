import React from 'react';
import {Subheading} from 'react-native-paper';
import {View, ViewStyle} from 'react-native';
import {ThemedIcon} from './ThemedIcon';
import {Screen} from './Screen';

const CONTAINER: ViewStyle = {
  flex: 1,
  justifyContent: 'center',
  alignItems: 'center',
  margin: 24,
};

export const EmptyPlaylist = () => {
  return (
    <Screen>
      <View style={CONTAINER}>
        <ThemedIcon name="disc-alert" size={80} />
        <Subheading style={{textAlign: 'center'}}>Empty playlists</Subheading>
      </View>
    </Screen>
  );
};
