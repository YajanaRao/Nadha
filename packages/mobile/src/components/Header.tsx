import React, {useState} from 'react';
import {Searchbar, useTheme} from 'react-native-paper';
import {View} from 'react-native';
import {useDispatch} from 'react-redux';
import {updateQuery} from '../actions/mediaStore';

export const Header = () => {
  const [query, setQuery] = useState('');
  const dispatch = useDispatch();

  const handleChange = (text: string) => {
    setQuery(text);
    dispatch(updateQuery(text));
  };

  const theme = useTheme();
  const {colors} = theme;

  return (
    <View style={{backgroundColor: colors.background}}>
      <Searchbar
        placeholder="Artists, songs or podcasts"
        onChangeText={handleChange}
        value={query}
        icon="magnify"
        style={{margin: 10}}
      />
    </View>
  );
};
