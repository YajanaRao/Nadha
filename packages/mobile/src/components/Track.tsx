import React from 'react';
import {List, useTheme} from 'react-native-paper';
import {StyleSheet, View} from 'react-native';

interface TrackProps {
  title: string;
  album?: string;
  artist?: string;
}

interface Props {
  track: TrackProps;
  active: boolean;
  play(): void;
}

export const Track = React.memo(({track, active, play}: Props) => {
  const theme = useTheme();
  const {colors} = theme;
  return (
    <View style={[styles.surface, {backgroundColor: colors.background}]}>
      <List.Item
        title={track.title}
        description={track.artist ? track.artist : track.album}
        right={(props) =>
          active ? (
            <List.Icon {...props} icon="poll" color={colors.accent} />
          ) : (
            // <List.Icon {...props} icon="more-vert" onPress={() => bs.current.snapTo(1)}/>
            false
          )
        }
        onPress={() => play()}
      />
    </View>
  );
});

const styles = StyleSheet.create({
  surface: {
    padding: 0,
    margin: 0,
    borderRadius: 4,
  },
});
