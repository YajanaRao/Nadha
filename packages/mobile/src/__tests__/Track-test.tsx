/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Track} from '../components/Track';

// Note: test renderer must be required after react-native.

it('In active track renders correctly', () => {
  const track = {title: 'title', artist: 'Tester'};
  const play = jest.fn();
  renderer.create(<Track track={track} active={false} play={play} />);
});

it('Active track renders correctly', () => {
  const track = {title: 'title', artist: 'Tester'};
  const play = jest.fn();
  renderer.create(<Track track={track} active play={play} />);
});

it('Blank component renders correctly', () => {
  const track = {title: 'title', album: 'Testing'};
  const play = jest.fn();
  const tree = renderer
    .create(<Track track={track} active play={play} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
