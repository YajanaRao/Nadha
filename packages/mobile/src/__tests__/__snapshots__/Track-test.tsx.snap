// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "margin": 0,
        "padding": 0,
      },
      Object {
        "backgroundColor": "#f6f6f6",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          title
        </Text>
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 14,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                },
                undefined,
              ],
            ]
          }
        >
          Testing
        </Text>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "marginRight": 0,
            },
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "#03dac4",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Ôêü
        </Text>
      </View>
    </View>
  </View>
</View>
`;
